uint8 new_leaf_id
---
uint8 exception_code